NebulaWorld/Factory/EntityManager.cs

```diff
@@ -0,0 +1,175 @@
+                }
+                if (prefab.isPowerConsumer) // this is actually essential to connect the consumer with the generators
+                {
+                    int entityId;
+                    if (GameMain.mainPlayer.factory.powerSystem.consumerCursor > 0)
+                    {
+                        entityId = GameMain.mainPlayer.factory.powerSystem.consumerPool[GameMain.mainPlayer.factory.powerSystem.consumerCursor - 1].entityId;
+                    }
+                    else
+                    {
+                        entityId = 0;
+                    }
+                    pcID = GameMain.mainPlayer.factory.powerSystem.NewConsumerComponent(entityId, prefab.workEnergyPerTick, prefab.idleEnergyPerTick);
+                }
+                if (prefab.minerType != EMinerType.None && prefab.minerPeriod > 0)
+                {
+                    // get veins that the miner could connect to
+                    int veinCount = 0;
+                    int[] tmp_ids = new int[1024];
+
+                    int[] veinIDs = MinerGetUsefullVeins(ref tmp_ids, ref veinCount, pos, rot, prefab.minerType);
+                    // veinIDs should now contain the id's the miner can connect to
+                    // now we need to add it to the miner pool and tell it the veins to connect to
+                    // entityId should be the last one in the array as we just added it
+                    int entityId = GameMain.mainPlayer.factory.entityPool[GameMain.mainPlayer.factory.entityCursor - 1].id;
+                    int minerId = GameMain.mainPlayer.factory.factorySystem.NewMinerComponent(entityId, prefab.minerType, prefab.minerPeriod);
+                    if (minerId != 0)
+                    {
+                        MinerComponent[] minerPool = GameMain.mainPlayer.factory.factorySystem.minerPool;
+                        minerPool[minerId].InitVeinArray(veinCount);
+                        if (veinCount > 0)
+                        {
+                            Array.Copy(veinIDs, minerPool[minerId].veins, veinCount);
+                        }
+                        for (int i = 0; i < minerPool[minerId].veinCount; i++)
+                        {
+                            GameMain.mainPlayer.factory.RefreshVeinMiningDisplay(minerPool[minerId].veins[i], entityId, 0);
+                        }
+                        minerPool[minerId].ArrageVeinArray();
+                        if (pcID != 0)
+                        {
+                            // this is hugely important to get power to the building!!!
+                            minerPool[minerId].pcId = pcID;
+                        }
+                        minerPool[minerId].GetMinimumVeinAmount(GameMain.mainPlayer.factory, GameMain.mainPlayer.factory.veinPool);
+                        // TODO: do some stuff with entitySignPool, is it important?
```

Comment:
This is basically updating the image overlays on top of rendered objects. In example, logistics stations, recipes in assembler, etc.